airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java

```diff
@@ -178,11 +180,14 @@ private static void writeStreamsWithNRecords(int minRecords,
       final CloseableQueue<byte[]> writeBuffer = writeBuffers.get(streamName);
       while (writeBuffer.size() > minRecords) {
         try {
-          final Stream<AirbyteRecordMessage> recordStream = Queues.toStream(writeBuffer)
```

Comment:
in practice we are always going to pull each batch fully into memory. i think we should change the `recordWriter` iface to accept list instead of stream, but don't want to bite that off right now.